feat(react-grid): improve column resizing (#2218)
DevExpress/devextreme-reactive: packages/dx-react-grid/docs/reference/table-column-resizing.md

defaultColumnWidths? | Array&lt;[TableColumnWidthInfo](#tablecolumnwidthinfo)&gt; | [] | Specifies initial column widths in uncontrolled mode.
onColumnWidthsChange? | (nextColumnWidths: Array&lt;[TableColumnWidthInfo](#tablecolumnwidthinfo)&gt;) => void | | Handles column width changes.
columnExtensions? | Array&lt;[TableColumnResizing.ColumnExtension](#tablecolumnresizingcolumnextension)&gt; | [] | Additional column properties that the plugin can handle.

> nextColumnResizing? | boolean | false | Specifies resizing mode

Reviewer: Is not clear for me

> resizingMode? | string | `widget` | Specifies resizing mode. May be defined as `widget` or `nextColumn`. In `widget` resizing mode not allow to use `auto` or `%` sizes.

Reviewer 2: ```suggestion
resizingMode? | string | `widget` | Specifies the resizing mode: `widget` or `nextColumn`. In `widget` mode, column widths cannot be specified using the `auto` and percentage values.
```

> resizingMode? | string | `widget` | Specifies the resizing mode: `widget` or `nextColumn`. Cannot use `auto` and percentage values to specify column widths in `widget` mode.

Reviewer 2: ```suggestion
resizingMode? | string | `widget` | Specifies the resizing mode: `widget` or `nextColumn`. In `widget` mode, `auto` and percentage values cannot be used to specify column widths.
```

> columnResizingMode? | string | 'widget' | Specifies resizing mode. May be defined as 'widget' or 'nextColumn'. In 'widget' resizing mode not allow to use 'auto' or '%' sizes.

Reviewer: I prefer to use this character ` instead of '. It highlights text in the site

Reviewer: Can we change the name to `resizingMode` because this plugin also has the `column` word in its name?

> resizingMode? | string | `widget` | Specifies the resizing mode: `widget` or `nextColumn`. In `widget` mode, column widths cannot be specified using the `auto` and percentage values.

Reviewer 3: ```suggestion
resizingMode? | string | `widget` | Specifies the resizing mode: `widget` or `nextColumn`. Cannot use `auto` and percentage values to specify column widths in `widget` mode.
```

Field | Type | Description
------|------|------------
columnName | string | A column name.

> width | number &#124; string | A column width. Allow to use `auto` or number with specified measurement units. Measurment units: `px`, `%`, `em`, `rem`, `vm`, `vh`, `vmin`,

Reviewer 2: ```suggestion
width | number &#124; string | A column width in pixels or CSS-accepted units (`auto`, `px`, `%`, `em`, `rem`, `vm`, `vh`, `vmin`).
```

-----|--------|------|------------
tableColumns | [Getter](../../../dx-react-core/docs/reference/getter.md) | Array&lt;[TableColumn](table.md#tablecolumn)&gt; | Table columns with new width values applied.
tableColumnResizingEnabled | [Getter](../../../dx-react-core/docs/reference/getter.md) | boolean | Specifies whether table column resizing is enabled.

> columnResizingMode | [Getter](../../../dx-react-core/docs/reference/getter.md) | string | Specifies resizing mode.

Reviewer 2: ```suggestion
columnResizingMode | [Getter](../../../dx-react-core/docs/reference/getter.md) | string | Specifies the resizing mode: `widget` or `nextColumn`.
```

changeTableColumnWidth | [Action](../../../dx-react-core/docs/reference/action.md) | ({ columnName: string, shift: number }) => void | Changes the specified column width. The column width is increased by the corresponding shift value, or decreased if the value is negative.

> draftTableColumnWidth | [Action](../../../dx-react-core/docs/reference/action.md) | ({ columnName: string, shift: number }) => void | Changes the specified column width used for preview. The column width is increased by the corresponding shift value, or decreased if the value is less than zero.

Reviewer 3: ```suggestion
draftTableColumnWidth | [Action](../../../dx-react-core/docs/reference/action.md) | ({ columnName: string, shift: number }) => void | Changes the specified column width used for preview. The column width is increased by the corresponding shift value or decreased if the value is less than zero.
```
